bigcache.go

```diff
@@ -137,6 +137,34 @@ func (c *BigCache) Get(key string) ([]byte, error) {
 	return shard.get(key, hashedKey)
 }
 
+// GetMulti reads entry for each of the keys.
+// It returns an ErrEntryNotFound when
+// no entry exists for the given key.
+func (c *BigCache) GetMulti(keys []string) ([][]byte, error) {
+
+    entries := make([][]byte,len(keys))
+    var err error
+    shardsLocked := make(map[uint64]bool)
+
+    for i := range keys{
+        hashedKey := c.hash.Sum64(keys[i])
+        shard := c.getShard(hashedKey)
```

Comment:
How about making it a bit more complex but at the same time limit number of locks. If we precompute `shard`s for all keys we can then get all required data from that shard with a single lock.

Reply:
that was my initial idea however i found that it was a lot slower initially, i will give it another go and see